### PR TITLE
Update rustic_hal to version 0.2

### PR DESCRIPTION
**Pull request by luckysori**

### CODE REVIEW

File: application/comit_node/src/http_api/rfc003/handlers/get_swaps.rs [hunk 1]
```diff
@@ -73,12 +73,11 @@ pub fn handle_get_swaps<T: MetadataStore<SwapId>, S: StateStore>(
...
                         };
 
-                        let mut hal_resource = HalResource::new(swap);
-                        hal_resource.with_link("self", swap_path(id));
+                        let hal_resource = HalResource::new(swap).with_link("self", swap_path(id));
 
                         for action in actions {
                             let link = new_action_link(&id, action);
-                            hal_resource.with_link(action, link);
+                            hal_resource.clone().with_link(action, link);
```
Comment (thomaseizinger) [inline application/comit_node/src/http_api/rfc003/handlers/get_swaps.rs]: :question: why do you need to clone here? Does `with_link` consume?

Looks like you would actually want to `fold` here maybe?

Comment (luckysori) [reply application/comit_node/src/http_api/rfc003/handlers/get_swaps.rs]: That's exactly what I need! Thank you.

File: application/comit_node/src/http_api/rfc003/handlers/get_swaps.rs [hunk 2]
```diff
@@ -73,13 +73,11 @@ pub fn handle_get_swaps<T: MetadataStore<SwapId>, S: StateStore>(
...
 
-                        let mut hal_resource = HalResource::new(swap);
-                        hal_resource.with_link("self", swap_path(id));
-
-                        for action in actions {
-                            let link = new_action_link(&id, action);
-                            hal_resource.with_link(action, link);
-                        }
+                        let hal_resource = HalResource::new(swap).with_link("self", swap_path(id));
+                        let hal_resource = actions.iter().fold(hal_resource, |acc, action| {
```
Comment (thomaseizinger) [inline application/comit_node/src/http_api/rfc003/handlers/get_swaps.rs]: Can you take ownership of `actions` here? Then you can do `into_iter` and avoid the deref of action in line 78.

File: application/comit_node/src/http_api/rfc003/routes.rs [hunk 3]
```diff
@@ -49,11 +49,11 @@ pub fn get_swap<T: MetadataStore<SwapId>, S: StateStore>(
 ) -> Result<impl Reply, Rejection> {
     handle_get_swap(&metadata_store, &state_store, id)
         .map(|(swap_resource, actions)| {
-            let mut response = HalResource::new(swap_resource);
-            for action in actions {
-                let link = new_action_link(&id, action);
-                response.with_link(action, link);
-            }
+            let response = HalResource::new(swap_resource);
+            let response = actions.iter().fold(response, |acc, action| {
```
Comment (thomaseizinger) [inline application/comit_node/src/http_api/rfc003/routes.rs]: Same thing here!

File: api_tests/npm_wrapper.sh [hunk 4]
```diff
@@ -7,9 +7,9 @@ export TEST_DIR="$1";
 export NVM_SH=$([ -e $NVM_DIR/nvm.sh ] && \
     echo "$NVM_DIR/nvm.sh" || echo /usr/local/opt/nvm/nvm.sh );
 . "$NVM_SH"
-nvm use 11.10.0 || (nvm install 11.10.0 && nvm use 11.10.0);
 
 cd "${ROOT}/api_tests";
+nvm use 11.10.0 || (nvm install 11.10.0 && nvm use 11.10.0);
```
Comment (D4nte) [inline api_tests/npm_wrapper.sh]: ❓  Is that still the latest version?

Comment (luckysori) [reply api_tests/npm_wrapper.sh]: All dealt with in #802.

**Pull request merged**